devimint/src/federation.rs

```diff
@@ -531,7 +539,171 @@ impl Fedimintd {
     }
 }
 
-pub async fn run_dkg(
+pub async fn run_cli_dkg(
+    params: HashMap<PeerId, ConfigGenParams>,
+    endpoints: BTreeMap<PeerId, String>,
+) -> Result<()> {
+    let auth_for = |peer: &PeerId| -> &ApiAuth { &params[peer].local.api_auth };
+
+    debug!(target: LOG_DEVIMINT, "Running DKG");
+    for endpoint in endpoints.values() {
+        poll("trying-to-connect-to-peers", || async {
+            crate::util::FedimintCli
+                .ws_status(endpoint)
+                .await
+                .context("dkg status")
+                .map_err(ControlFlow::Continue)
+        })
+        .await?;
+    }
+
+    debug!(target: LOG_DEVIMINT, "Connected to all peers");
+
+    for (peer_id, endpoint) in &endpoints {
+        let status = crate::util::FedimintCli.ws_status(endpoint).await?;
+        assert_eq!(
+            status.server,
+            ServerStatus::AwaitingPassword,
+            "peer_id isn't waiting for password: {peer_id}"
+        );
+    }
+
+    debug!(target: LOG_DEVIMINT, "Setting passwords");
+    for (peer_id, endpoint) in &endpoints {
+        crate::util::FedimintCli
+            .set_password(auth_for(peer_id), endpoint)
+            .await?;
+    }
+    let (leader_id, leader_endpoint) = endpoints.iter().next().context("missing peer")?;
+    let followers = endpoints
+        .iter()
+        .filter(|(id, _)| *id != leader_id)
+        .collect::<BTreeMap<_, _>>();
+
+    debug!(target: LOG_DEVIMINT, "calling set_config_gen_connections for leader");
+    let leader_name = "leader".to_string();
+    crate::util::FedimintCli
+        .set_config_gen_connections(auth_for(leader_id), leader_endpoint, &leader_name, None)
+        .await?;
+
+    crate::util::FedimintCli
+        .get_default_config_gen_params(auth_for(leader_id), leader_endpoint)
+        .await?; // sanity check
+
+    let server_gen_params = &params[leader_id].consensus.modules;
```

Comment:
might want to check that the schemas are compatible between the default params and what we defined